### PR TITLE
Use standard method for KeepScreenOn flag

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: src/core/androidplatformutilities.cpp [hunk 1]
```diff
@@ -260,6 +260,7 @@ bool AndroidPlatformUtilities::checkAndAcquirePermissions( const QString &permis
 
 void AndroidPlatformUtilities::setScreenLockPermission( const bool allowLock )
 {
+#if 0
```
Comment (m-kuhn) [inline src/core/androidplatformutilities.cpp]: Hmm... this ifdef doesn't help to make it any better :D

**Comment by nirvn** [comment]
I tried something similar when I first implemented this, and it didn't work because the java launcher wasn't persistent. 

It wasn't exactly like that so maybe your method works. Did you give it a try on your smartphone for a minute or two?

**Comment by m-kuhn** [comment]
TBH, I don't completely remember. I had issues with a locally compiled version using the previous approach which made me switch to this but then I abandoned the patch and just found it again now.

I was hoping @signedav could comment on it since he is apparently having similar issues given his recent accidentally commented out code.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-a35d1fb1c5451099016eab7c5f60200bf8d257c0-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-a35d1fb1c5451099016eab7c5f60200bf8d257c0-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-a35d1fb1c5451099016eab7c5f60200bf8d257c0-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-a35d1fb1c5451099016eab7c5f60200bf8d257c0-x86_64.apk)

**Comment by signedav** [comment]
I suggested a similar implementation here, that worked #833 But since it had no issues on the APKs created here, we did not merge it.

Should I test it with this part @m-kuhn https://github.com/opengisch/QField/pull/907/files#diff-6105cfd3416184db1bb8ab6988115440R99 ?

**Comment by m-kuhn** [comment]
I don't trust this one a lot. The runOnMainThread from #833 looks better. But the discussion in there was based on a mix between different issues. What we are trying to solve here are crashes caused by building locally, nothing related to #833.

If (a clean version of) #833 works for travis builds *and* for local builds, I'd be in favor of merging it.

**Comment by nirvn** [comment]
I think you're settling for the right approach.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-57a3910d79e059454814c5c2fd08700d06a592fe-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-57a3910d79e059454814c5c2fd08700d06a592fe-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-57a3910d79e059454814c5c2fd08700d06a592fe-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-57a3910d79e059454814c5c2fd08700d06a592fe-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-5667e57eba2e52c8736d666262efa6bfd3a4b0a4-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-5667e57eba2e52c8736d666262efa6bfd3a4b0a4-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-5667e57eba2e52c8736d666262efa6bfd3a4b0a4-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-907-5667e57eba2e52c8736d666262efa6bfd3a4b0a4-x86_64.apk)

**Comment by signedav** [comment]
I set it backport. In case that someone want's to test or debug on the backport branch, okay?

**Comment by m-kuhn** [comment]
@signedav did you test the apk?

**Comment by signedav** [comment]
Yes.

**Pull request merged**